workflows/cloudify_system_workflows/snapshot.py

```diff
@@ -0,0 +1,574 @@
+    client = get_rest_client()
+    defaults = _get_broker_configuration(client)
+    defaults['version'] = _VERSION
+    result = {}
+    for deployment in client.deployments.list():
+        deployment_result = {}
+        for node in client.nodes.list(deployment_id=deployment.id):
+            if _is_compute(node):
+                node_result = {}
+                for node_instance in client.node_instances.list(
+                        deployment_id=deployment.id,
+                        node_name=node.id):
+                    overrides = {}
+                    current = node_instance.runtime_properties.get(
+                        'cloudify_agent', {})
+                    for k, v in defaults.iteritems():
+                        overrides[k] = current.get(k, v)
+                    node_result[node_instance.id] = overrides
+                deployment_result[node.id] = node_result
+        result[deployment.id] = deployment_result
+    with open(os.path.join(tempdir, _AGENTS_FILE), 'w') as out:
+        out.write(json.dumps(result))
+
+
+def _update_es_node(es_node):
+    if es_node['_type'] == 'deployment':
+        workflows = es_node['_source']['workflows']
+        if 'install_new_agents' not in workflows:
+            workflows['install_new_agents'] = {
+                'operation': 'cloudify.plugins.workflows.install_new_agents',
+                'parameters': {
+                    'install_agent_timeout': {
+                        'default': 300
+                    },
+                    'node_ids': {
+                        'default': []
+                    },
+                    'node_instance_ids': {
+                        'default': []
+                    }
+                },
+                'plugin': 'default_workflows'
+            }
+    if es_node['_type'] == 'node':
+        source = es_node['_source']
+        type_hierarchy = source.get('type_hierarchy', [])
+        if 'cloudify.nodes.Compute' in type_hierarchy:
+            operations = source['operations']
+            op_name = 'cloudify.interfaces.cloudify_agent.create_amqp'
+            if op_name not in operations:
+                operations[op_name] = {
+                    'inputs': {
+                        'install_agent_timeout': 300
+                    },
+                    'has_intrinsic_functions': False,
+                    'plugin': 'agent',
+                    'retry_interval': None,
+                    'max_retries': None,
+                    'executor': 'central_deployment_agent',
+                    'operation': 'cloudify_agent.operations.create_agent_amqp'
+                }
+
+    if es_node['_type'] == 'blueprint':
+        source = es_node['_source']
+        if 'description' not in source:
+            source['description'] = ''
+
+
+def _restore_elasticsearch(ctx, tempdir, es, metadata):
+    ctx.send_event('Deleting all ElasticSearch data')
+    elasticsearch.helpers.bulk(
```

Comment:
We have already implemented that validation (plus flag to control it) but it will be pushed as separate pull request.